aptos-move/block-executor/src/view.rs

```diff
@@ -1429,11 +1429,28 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>> LatestView<'a, T, S> {
         layout: UnknownOrLayout,
         kind: ReadKind,
     ) -> PartialVMResult<ReadResult> {
-        debug_assert!(
-            !state_key.is_module_path(),
-            "Reading a module {:?} using ResourceView",
-            state_key,
-        );
+        let msg = if state_key.is_module_path() {
```

Comment:
This calls
```rust
pub fn is_code(&self) -> bool {
  matches!(self.get_path(), Path::Code(_))
}
```
with
```rust
 pub fn get_path(&self) -> Path {
    bcs::from_bytes::<Path>(&self.path).expect("Unexpected serialization error")
}
```
isn't this expensive performance-wise?

In theory, when we do `StateKey::module(...)` we can return a wrapper `ModuleKey(StateKey)` which is only constructed via state key interface, and same for resource group / resource so we differentiate keys with the type system? This way we do not pay path deserialization costs, but we can do this if we see this actually making numbers worse I guess